### PR TITLE
feat: Add suggestions for no-console

### PR DESCRIPTION
**Pull request by edi9999**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

This adds suggestions both for "no-console".

### CODE REVIEW

File: lib/rules/no-console.js [hunk 1]
```diff
@@ -105,7 +108,15 @@ module.exports = {
             context.report({
                 node,
                 loc: node.loc,
-                messageId: "unexpected"
+                messageId: "unexpected",
+                suggest: [
+                    {
+                        messageId: "removeConsole",
+                        fix(fixer) {
+                            return fixer.removeRange(node.parent.parent.range);
```
[the remark this answers is not in the document]
Comment (ota-meshi) [reply lib/rules/no-console.js:116]: I think the code below may cause a syntax error due to the removal of `console.x()` even if only the ExpressionStatement node is removed.

```js
if (a) console.log(b)
label: console.log(b)
```

↓

```js
if (a) 
label:
```

I think we need to further check the parent node to see if it is one of the four.

```js
console.log(b) // Program
switch (a) {
    case 1: console.log(b) // SwitchCase
}
if (a) {
  console.log(b) // BlockStatement
}
class A {
  static {
    console.log(b) // StaticBlock
  }
}
```

[the remark this answers is not in the document]
Comment (mdjermanovic) [reply lib/rules/no-console.js:116]: > I think we need to further check the parent node to see if it is one of the four.

👍 

We can use [`astUtils.STATEMENT_LIST_PARENTS`](https://github.com/eslint/eslint/blob/55c1685aab0fa75074ac1033abfa6e34cb7baaf6/lib/rules/utils/ast-utils.js#L40C7-L40C29) for this check.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]
@edi9999 proposal for this feature is now accepted. Can you take a look at https://github.com/eslint/eslint/pull/17542#discussion_r1319839007?

**Comment by domnantas** [comment]
Will this auto-remove console.* when the rule is set to `"no-console": "warn"` and autofix is enabled in the editor?

**Comment by Rec0iL99** [comment]
@edi9999 friendly ping. Do you plan on finishing up this pr? 

There are a few review comments for you to look into. Thanks.

**Comment by edi9999** [comment]
I would like to do it, but I see that there seem to be quite a few cornercases, and my work is pretty intense at the current time.

**Comment by Rec0iL99** [comment]
Since this has been marked stale again, can I give this a try? I'll try sending in a PR by next weekend :)

**Comment by edi9999** [comment]
@Rec0iL99 absolutely !

**Comment by mdjermanovic** [comment]
Alright, then I'm closing this and assigning https://github.com/eslint/eslint/issues/17493 to @Rec0iL99.

**Pull request closed without merging**